### PR TITLE
Add config option to trim package name from grpc server span resource name

### PR DESCRIPTION
**Pull request by tylerbenson**
Disabled by default.  To enable:
```
-Ddd.trace.grpc.server.trim-package-resource=true
```

### CODE REVIEW

File: dd-java-agent/instrumentation/grpc-1.5/src/main/java/datadog/trace/instrumentation/grpc/client/GrpcClientDecorator.java [hunk 1]
```diff
@@ -32,6 +33,12 @@ protected String service() {
     return null;
   }
 
+  public <RespT, ReqT> AgentSpan onCall(
+      final AgentSpan span, final MethodDescriptor<ReqT, RespT> method) {
+    span.setResourceName(method.getFullMethodName());
+    return span;
+  }
```
Comment (richardstartin) [inline dd-java-agent/instrumentation/grpc-1.5/src/main/java/datadog/trace/instrumentation/grpc/client/GrpcClientDecorator.java:40]: Can't this just be done in place?

File: dd-java-agent/instrumentation/grpc-1.5/src/main/java/datadog/trace/instrumentation/grpc/client/TracingClientInterceptor.java [hunk 2]
```diff
@@ -37,9 +37,9 @@
     }
 
     final AgentSpan span = startSpan(GRPC_CLIENT).setMeasured(true);
-    span.setResourceName(method.getFullMethodName());
     try (final AgentScope scope = activateSpan(span)) {
       DECORATE.afterStart(span);
+      DECORATE.onCall(span, method);
```
Comment (richardstartin) [inline dd-java-agent/instrumentation/grpc-1.5/src/main/java/datadog/trace/instrumentation/grpc/client/TracingClientInterceptor.java:42]: KISS: just set the resource name here.

File: dd-java-agent/instrumentation/grpc-1.5/src/main/java/datadog/trace/instrumentation/grpc/server/GrpcServerDecorator.java [hunk 3]
```diff
@@ -1,18 +1,49 @@
...
+        @Override
+        public String apply(String fullName) {
+          int index = fullName.lastIndexOf(".");
+          if (index > 0) {
+            return fullName.substring(index + 1);
+          } else {
+            return fullName;
+          }
+        }
+      };
```
Comment (richardstartin) [inline dd-java-agent/instrumentation/grpc-1.5/src/main/java/datadog/trace/instrumentation/grpc/server/GrpcServerDecorator.java:35]: nit: we wouldn't need a helper class if we just implemented `Function<String, String>` here. On the one hand, that's obviously gross and I wouldn't do it in application code ever, but I don't really see the instrumentations the same way as normal code, the unattainable ideal is zero classes and zero bytecode.

**Review by richardstartin — APPROVED** [review]

File: dd-java-agent/instrumentation/grpc-1.5/src/main/java/datadog/trace/instrumentation/grpc/server/GrpcServerDecorator.java (hunk 3 above)
Comment (tylerbenson) [reply dd-java-agent/instrumentation/grpc-1.5/src/main/java/datadog/trace/instrumentation/grpc/server/GrpcServerDecorator.java:35]: Do you really want me to do that?  I agree it seems gross.

File: dd-java-agent/instrumentation/grpc-1.5/src/main/java/datadog/trace/instrumentation/grpc/client/GrpcClientDecorator.java (hunk 1 above)
Comment (tylerbenson) [reply dd-java-agent/instrumentation/grpc-1.5/src/main/java/datadog/trace/instrumentation/grpc/client/GrpcClientDecorator.java:40]: I was trying to keep the client/server consistent.  If this bothers you I can revert this part.

File: dd-java-agent/instrumentation/grpc-1.5/src/main/java/datadog/trace/instrumentation/grpc/server/GrpcServerDecorator.java (hunk 3 above)
Comment (richardstartin) [reply dd-java-agent/instrumentation/grpc-1.5/src/main/java/datadog/trace/instrumentation/grpc/server/GrpcServerDecorator.java:35]: Hey, nowhere near as gross as the whole decorator hierarchy thing ;)

**Review by bantonsson — APPROVED** [review]

**Pull request merged**